pkgs/development/rocm-modules/6/llvm/base.nix

```diff
@@ -0,0 +1,175 @@
+{ lib
+, stdenv
+, gcc12Stdenv
+, fetchFromGitHub
+, rocmUpdateScript
+, pkg-config
+, cmake
+, ninja
+, git
+, doxygen
+, sphinx
+, lit
+, libxml2
+, libxcrypt
+, libedit
+, libffi
+, mpfr
+, zlib
+, ncurses
+, python3Packages
+, buildDocs ? true
+, buildMan ? true
+, buildTests ? true
+, targetName ? "llvm"
+, targetDir ? "llvm"
+, targetProjects ? [ ]
+, targetRuntimes ? [ ]
+, llvmTargetsToBuild ? [ "NATIVE" ] # "NATIVE" resolves into x86 or aarch64 depending on stdenv
+, extraPatches ? [ ]
+, extraNativeBuildInputs ? [ ]
+, extraBuildInputs ? [ ]
+, extraCMakeFlags ? [ ]
+, extraPostPatch ? ""
+, checkTargets ? [(
+  lib.optionalString buildTests (
+    if targetDir == "runtimes"
+    then "check-runtimes"
+    else "check-all"
+  )
+)]
+, extraPostInstall ? ""
+, hardeningDisable ? [ ]
+, requiredSystemFeatures ? [ ]
+, extraLicenses ? [ ]
+, isBroken ? false
+}:
+
+let stdenv' = stdenv; in
+let stdenv =
+      if stdenv'.cc.cc.isGNU or false && lib.versionAtLeast stdenv'.cc.cc.version "13.0"
+      then gcc12Stdenv
+      else stdenv';
+in
+
```

Comment:
Should this still be here? Doesn't rocm-llvm 6 build with gcc 13?

Reply:
Good catch. I will verify that it builds with gcc 13 and remove this if it does.

Reply:
I have a good machine for building things so let me know if you want me to test something

Reply:
> I have a good machine for building things so let me know if you want me to test something

Thanks for the offer. I have 128 GB of RAM now as well, and that's enough for me to build ROCm now.

Reply:
It still does not build with GCC 13.
What fails to build without `gcc12Stdenv` is `rStdenv` and `runtimes` inside `llvm/default.nix`.

I was thinking about pushing a change to reduce the usages down to strictly those, but I have the feeling that better to not have more than one specific version of GCC involved there.